### ISSUE TITLE
Log files are not stored at the chosen path

### ISSUE THREAD

**Issue by bennypi** [flathub/de.bund.ausweisapp.ausweisapp2#2]
Hi there, 

when I want to save the log file, the native file chooser is shown, with my home directory as the current directory. After I enter a file name for the log and click on save, a message appears that the log is saved. The file however is not saved at the location that I chose in the file chooser, but somwhere deep in the file system. See the attached screenshot for an example.
[
![Bildschirmfoto vom 2020-12-10 15-14-09](https://user-images.githubusercontent.com/2631090/101785234-cff45d00-3afc-11eb-9424-0e0d0f62c64f.png)
](url)
I am using Linux Mint 20.

This behavior is really confusing and should be fixed.

**Comment by e-user** [comment]
This appears to be an upstream bug. I checked and AusweisApp2 appears to always to always store the log under `${XDG_RUNTIME_DIR}/doc/${UNKNOWN_ID}/${FILENAME}` regardless of the path chosen, only `${FILENAME}` is honored.

@bennypi would you mind reporting this to https://www.ausweisapp.bund.de/en/help-and-support/support-error-reports/ ?

**Comment by misery** [comment]
The AusweisApp2 uses the FileDialog of Qt.
https://github.com/Governikus/AusweisApp2/blob/1.22.0/resources/qml/%2Bdesktop/main.qml#L154

There should be no restriction of AusweisApp2. If I run this Flatpak on ArchLinux I can save the logfile to selected directory. Just the shown message contains the wrong path.

Does this flatpak uses Qt's flatpak plugin?
https://code.qt.io/cgit/qt/qtbase.git/log/?qt=grep&q=flatpak

**Comment by e-user** [comment]
@bennypi @misery could you please install the version from `flatpak install --user https://dl.flathub.org/build-repo/33144/de.bund.ausweisapp.ausweisapp2.flatpakref` and run it using `flatpak run de.bund.ausweisapp.ausweisapp2//test` to check whether you can save to ~/Documents? The reported filename is still wrong but that's really an issue with Qt, the portal code and how AusweisApp2 logs the path - the file should still land in the right location.

**Comment by misery** [comment]
Thanks, it works for me. But saving was working for me before that, too. ;-) Just the displayed path is strange.
Looks like this bug: https://bugreports.qt.io/browse/QTBUG-77825

**Comment by e-user** [comment]
@misery thank you for testing. 

I had already checked the bug report you linked and also tested the sample project linked from the report and concluded this is a different issue.

The actual issue stems from the use of the Flatpak file chooser portal, documented at https://flatpak.github.io/xdg-desktop-portal/portal-docs.html: 

```
Description
The FileChooser portal allows sandboxed applications to ask the user for access to files outside the sandbox. The portal backend will present the user with a file chooser dialog.

The selected files will be made accessible to the application via the document portal, and the returned URI will point into the document portal fuse filesystem in /run/user/$UID/doc/.
```

So the Qt application tries to access a file outside its sandbox and (correctly) requests access through the document portal, which returns a path into the aforementioned fuse filesystem, which correspond to the paths we're seeing. 

There are a couple of related bug reports for other applications:
* https://github.com/flatpak/flatpak/issues/1307
* https://github.com/flatpak/xdg-desktop-portal/issues/141

In conclusion, this seems to be a design issue with the document portal implementation. 

I just built and installed with `--filesystem=home` and that fixes the problem as people suggested on those other issues, but it's not a nice fix but a bad workaround that's detrimental to security. Since it's a Free Software application, I think we can give it a pass and I'll provide another build for you to test with that permission.

**Comment by e-user** [comment]
`flatpak install --user https://dl.flathub.org/build-repo/33164/de.bund.ausweisapp.ausweisapp2.flatpakref` doesn't display the erroneous behavior anymore for me.

@bennypi could you please verify?
